test/visual/components/index.spec.ts

```diff
@@ -43,25 +44,116 @@ test('showing grid while resizing elements', async ({ page, argosScreenshot }) =
 
   await editorModel.waitForOverlay();
 
-  const firstInput = editorModel.appCanvas.locator('input').first();
+  const input = editorModel.appCanvas.locator('input').first();
 
-  await clickCenter(page, firstInput);
+  await clickCenter(page, input);
 
-  const firstInputBoundingBox = await firstInput.boundingBox();
+  const inputBoundingBox = await input.boundingBox();
 
   await page.mouse.move(
-    firstInputBoundingBox!.x + firstInputBoundingBox!.width - 5,
-    firstInputBoundingBox!.y + firstInputBoundingBox!.height / 2,
+    inputBoundingBox!.x + inputBoundingBox!.width - 5,
+    inputBoundingBox!.y + inputBoundingBox!.height / 2,
     { steps: 10 },
   );
 
   await page.mouse.down();
 
   await page.mouse.move(
-    firstInputBoundingBox!.x + firstInputBoundingBox!.width / 2,
-    firstInputBoundingBox!.y + firstInputBoundingBox!.height / 2,
+    inputBoundingBox!.x + inputBoundingBox!.width / 2,
+    inputBoundingBox!.y + inputBoundingBox!.height / 2,
     { steps: 10 },
   );
 
   await argosScreenshot('resize-grid');
 });
+
+test('showing drag-and-drop previews', async ({ page, argosScreenshot }) => {
+  const editorModel = new ToolpadEditor(page);
+  await editorModel.goToPageById('8ixPqyI');
+
+  await editorModel.waitForOverlay();
+
+  const appCanvasBoundingBox = await editorModel.appCanvas.locator('body').boundingBox();
+
+  const screenshotConfig = {
+    clip: appCanvasBoundingBox || undefined,
+  };
+
+  const getDropPreviewLeftCoordinates = (boundingBox: BoundingBox): [number, number] => [
+    boundingBox!.x + (1 / 3) * boundingBox!.width,
```

Comment:
I don't believe any of these no-null assertion operators are necessary
```suggestion
    boundingBox.x + (1 / 3) * boundingBox.width,
```